File: web/app/client/app/spaces/projects.js
app.spaces.projects = controller => (a,x) => [
  app.close( controller ),
  a.h1('Projects'),
  controller.routes( {
    '/?': app.spaces.projects.index,
    '/~new': app.spaces.projects.new,
    '/:project_id*': app.spaces.projects.project,
  } ),
]

File: web/app/api/models/spaces/projects.rb
module App
  class Api
    module Models
      module Spaces
        class Projects < Models::SpacesCollection

          def to_a
            Spaces.universe.projects.identifiers.sort
          end

          def by(identifier)
            Project.new(identifier)
          end

          def create(project)
            by(project[:identifier]).create
          end

        end
      end
    end
  end
end
